Add usb device info script

kb100 opened the pull request:
I think that a lot of people will find this script useful. Feel free to make pre-merge suggestions, but short of unforseen bugs I believe it is ready as is.

Zopieux commented on usb/usb.py:
```diff
@@ -0,0 +1,245 @@
+
+###############################################################################
+# END CONFIG
+# DO NOT EDIT ANYTHING AFTER THIS POINT UNLESS YOU ARE A DEVELOPER
```
UNLESS YOU *KNOW WHAT YOU ARE DOING* :wink:

Zopieux commented on usb/usb.py:
```diff
@@ -0,0 +1,245 @@
+def getLockedCryptOutput(path):
+    form = "<span color='{}'>[{} {}]</span>"
+    kname = getKernelName(path)
+    output = form.format(LOCKED_COLOR, FA_LOCK, kname)
```
`pangoEscape` is not used here. Is it on purpose?

kb100 replied:
AFAIK `df -h --output=avail` will never output any character that needs to be escaped ( '&', '<', '>'). Should I guard against this anyway?

Zopieux commented on usb/usb.py:
```diff
@@ -0,0 +1,245 @@
+    parentKernelName = getParentKernelName(path)
+    
+    block = "<span color='{}'>[{} {}:{}]</span>"
+    block = block.format(color, FA_UNLOCK, parentKernelName, kernelName)
```
Same remark here on `pangoEscape`.

kb100 replied:
As mentioned above `getSpaceAvailable` uses `df -h --output=avail` which I don't believe needs to be escaped. However, I could see how escaping just in case might be wanted. This is where I would do it. Also as far as I know a kernel name will never have bad characters  in it either. You know as I try to explain these things I feel like guarding against bad characters should be done anyway.

Zopieux replied:
I do agree it is very unlikely these strings will ever contain full HTML tags. Nevertheless, even though these functions *should* be safe, the call to `pangoEscape` is basically free.

Zopieux commented on usb/usb.py:
```diff
@@ -0,0 +1,245 @@
+def getDiskWithNoPartitionsOutput(path):
+    form = "<span color='{}'>[{}] {}</span>"
+    kernelName = getKernelName(path)
+    return form.format(PARTITIONLESS_COLOR, kernelName, PARTITIONLESS_TEXT)
```
Same remark here on `pangoEscape`.

Zopieux commented on usb/usb.py:
```diff
@@ -0,0 +1,245 @@
+    return { path : makeAttributeMap(path) for path in paths}
+
+leaves = getLeafDevicePaths()
+leaves = list(filter(lambda path: not fastIgnore(path), leaves))
```
May I suggest `leaves = [path for path in leaves if not fastIgnore(path)]` with removes both list and filter (and the lambda)?

Zopieux commented on usb/usb.py:
```diff
@@ -0,0 +1,245 @@
+leaves = getLeafDevicePaths()
+leaves = list(filter(lambda path: not fastIgnore(path), leaves))
+attributeMaps = getAttributeMaps(leaves)
+leaves = filter(lambda path: not ignore(path, attributeMaps[path]), leaves) 
```
`leaves = (path for path in leaves if not ignore(path, attributeMaps[path]))`

Zopieux commented:
Thanks for the contribution! The Python script is well formatted and documented. I suggested a few things but it is mainly *aesthetics*, feel free to use them or not.

kb100 commented:
@Zopieux I've addressed all of your concerns in https://github.com/kb100/i3blocks-contrib/commit/f4645ca4b5bfda26e6ef06a1ec5593009e8640a9, what do you think of it now?

kb100 commented:
I've squashed all the commits into one and made the title of the commit match @vivien's suggestions.

Zopieux commented:
Great! I don't have any further suggestions.

vivien commented:
Thanks to both of you!
